docs/flowcharts/TestExecution.md

```diff
@@ -0,0 +1,50 @@
+---
+id: testexecution
+title: Test Execution
+---
+This flowchart explains the test execution process and the interaction between @wdio/runner and the most of the other WebdriverIO packages.
+<div>
+    <div class="flowcharttogglemenu">
+        <span>[</span>
+        <a class="flowcharttogglelink" onclick="createFlowChart(setupTest)">Setup test</a>
+        <span>|</span>
+        <a class="flowcharttogglelink" onclick="createFlowChart(runTest)">Run test</a>
+        <span>]</span>
+    </div>
+    <div id="flowChartGraphDiv" style="color: blue; margin-left: auto; margin-right: auto; padding-left: auto; padding-right: auto; text-align: center;"></div>
+</div>
+
+<script src="https://unpkg.com/mermaid@8.3.1/dist/mermaid.min.js"></script>
+<script src="/js/flowchart.js"></script>
+<script>
+    var setupTest = `
+    graph TD
+        START("@wdio/runner index.js called from<br>child process via a run message.")-->
+        EXECUTERUN["Execute @wdio/runner index.js run method."]-->
+        PARSECONFIG["Parse config file using @wdio/config configParser<br>method. Add to @wdio/runner index.js class."]-->
+        MERGECLIARGS[Merge CLI args and host<br>port changes into config]-->
+        REMOVESERVICES["Use @wdio/config filterWorkerServices<br>to remove services not used in worker"]-->
+        SETLOGLEVEL[Set log level]-->
+        SETISMULTIREMOTEFLAG[Set isMultiremote flag]-->
```

Comment:
I am going to try and add a branch here (between lines 28 and 29) that describes the process of creating the WebdriverIO monad. This is missing from the flowcharts and is quite important! :)

This branch should be able to explain the interaction between @wdio/runner and webdriverio/webdriver/@wdio/utils, @wdio/sync.

Reply:
Hello I am still working on this ticket, unfortunately the task is a lot harder than I originally thought.  To create the flow charts, I am stepping through the code using the VSCode debugger then making notes for the most important functions. 

I am posting my VSCode debugger setup because I feel **stepping through the code** is the easiest way to **grok** the system.  Also I am not  100% certain flowcharts will be the best solution for getting up to speed because they could become outdated quickly. 

I will try to make them very simple and easy to update.  Hope to complete this task soon.

**Setting up debugger with VSCode**
1. Add `--inspect-brk` to  ./node_modules/.bin/wdio.cmd (or wdio for linux). 
2. Add the below configuration to your local VSCode launch.js. 

**wdio.cmd**
```
@IF EXIST "%~dp0\node.exe" (
  "%~dp0\node.exe"  "%~dp0\..\@wdio\cli\bin\wdio.js" %*
) ELSE (
  @SETLOCAL
  @SET PATHEXT=%PATHEXT:;.JS;=;%
  node --inspect-brk  "%~dp0\..\@wdio\cli\bin\wdio.js" %*
)
```

**launch.js**
```
{
    "version": "0.2.0",
    "configurations": [
        {
           "type": "node",
            "request": "launch",
            "protocol": "inspector",
            "port": 9229,
            "name": "WebdriverIO",
            "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/wdio",
            "windows": {
                "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/wdio.cmd"
            },
            "timeout": 100000000,
            "cwd": "${workspaceRoot}",
            "console": "integratedTerminal",
            "args":[
                "--mochaOpts.timeout=99999999",
                "--spec", "${workspaceRoot}/<path to test>",
                "${workspaceRoot}/wdio.conf.js"
            ],
        }
    ]
}
```

James Davis